tensorflow/python/ops/array_ops.py

```diff
@@ -575,6 +579,16 @@ def _slice_helper(tensor, slice_spec, var=None):
     TypeError: If the slice indices aren't int, slice, ellipsis,
       tf.newaxis or scalar int32/int64 tensors.
   """
+  if isinstance(slice_spec, np.ndarray):
+    ndims_slice_spec = slice_spec.ndim
+    if ndims_slice_spec != 0:
```

Comment:
I think calling `boolean_mask()` should be dependent on a boolean dtype, not shape. `x[x > a]` is perfect well defined for boolean tensors, even though tf.boolean_mask currently does not support scalars for some reason. Consider how NumPy handles this:
```
In [6]: import numpy as np

In [7]: x = np.array(1)

In [8]: x[x > 0]
Out[8]: array([1])
```

Reply:
I think @shoyer what you said make sense, indeed, boolean_mask() should be not dependent on shape, so the easiest change can be:

  if isinstance(slice_spec, (np.ndarray, ops.Tensor)):
      return boolean_mask(tensor=tensor, mask=slice_spec)

Also, I am going to change the array_ops_test.py accordingly. Will such change be Okay?

In terms of boolean dtype I noticed that boolean_mask() does not run any boolean dtype check, so I believe the above change is consistent with boolean_mask().

Reply:
Yes, please make this change.

Reply:
@alextp, before I submit my latest PR request, it seems that there are plenty of commits made by other contributors, then I tried 

```
git reset HEAD~2
git push -f origin boolean_mask_numpy 
```
my push got messed up. Could you help me with that? Thank you very much.